core/src/test/java/org/elasticsearch/index/query/functionscore/ScoreFunctionBuilderTests.java

```diff
@@ -0,0 +1,146 @@
+
+        try {
+            new FieldValueFactorFunctionBuilder(null);
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new FieldValueFactorFunctionBuilder("").modifier(null);
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new GaussDecayFunctionBuilder(null, "", "", "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new GaussDecayFunctionBuilder("", "", null, "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new GaussDecayFunctionBuilder("", "", null, "", randomIntBetween(1, 100));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new GaussDecayFunctionBuilder("", "", null, "", randomIntBetween(-100, -1));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new LinearDecayFunctionBuilder(null, "", "", "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new LinearDecayFunctionBuilder("", "", null, "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new LinearDecayFunctionBuilder("", "", null, "", randomIntBetween(1, 100));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new LinearDecayFunctionBuilder("", "", null, "", randomIntBetween(-100, -1));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new ExponentialDecayFunctionBuilder(null, "", "", "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new ExponentialDecayFunctionBuilder("", "", null, "");
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new ExponentialDecayFunctionBuilder("", "", null, "", randomIntBetween(1, 100));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
+        try {
+            new ExponentialDecayFunctionBuilder("", "", null, "", randomIntBetween(-100, -1));
+            fail("must not be null");
+        } catch(IllegalArgumentException e) {
+            //all good
+        }
+
```

Comment:
Yeah. I can see there are lots of integration tests for each function but it would be good to build out unit tests for them too. As I said, it doesn't need to be done in this PR but it's something we should do IMO